src/pyodide/internal/snapshot.ts

```diff
@@ -102,6 +103,46 @@ function loadDynlib(
   }
 }
 
+/**
+ * This function is used to ensure the order in which we load SO_FILES stays the same.
+ *
+ * The sort always puts _lzma.so and _ssl.so
+ * first, because these SO_FILES are loaded in the baseline snapshot, and if we want to generate
+ * a package snapshot while a baseline snapshot is loaded we need them to be first. The rest of the
+ * files are sorted alphabetically.
+ *
+ * The `filePaths` list is of the form [["folder", "file.so"], ["file.so"]], so each element in it
+ * is effectively a file path.
+ */
+function sortSoFiles(filePaths: FilePath[]): FilePath[] {
+  const result = [];
+  let hasLzma = false;
+  let hasSsl = false;
+  const lzmaFile = '_lzma.so';
+  const sslFile = '_ssl.so';
+  for (const path of filePaths) {
+    if (path.length == 1 && path[0] == lzmaFile) {
+      hasLzma = true;
+    } else if (path.length == 1 && path[0] == sslFile) {
+      hasSsl = true;
+    } else {
+      result.push(path);
+    }
+  }
+
+  // JS might handle sorting lists of lists fine, but I'd rather be explicit here and make it compare
+  // strings.
+  result.sort((a, b) => a.join('/').localeCompare(b.join('/')));
```

Comment:
This is a really poor performance sort since it creates new strings on every comparison. Either do:
```js
result.map(x => x.join('/')).sort(...).map(x => x.split('/'));
```
or even better performance would be to use:
```js
function pathCompare(p, q) {
  for (let idx = 0; idx < p.length && x < q.length; idx++) {
    let cmp = p[idx].localeCompare(q[idx]);
    if (cmp) {
      return cmp;
    }
  }
  return p.length - q.length;
}
result.sort(pathCompare);
```